cmd/send/send.go

```diff
@@ -51,5 +52,16 @@ func parseArgumentsAndCallSend(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("getting stage artifacts root failed, original error: '%w'", err)
 	}
 
-	return sendImplementation.PrepareAndSendSubmissionInformationPackage(common_flags.KafkaEndpoints, transferTopicName, stageArtifactsRoot)
+	err = sendImplementation.PrepareAndSendSubmissionInformationPackage(common_flags.KafkaEndpoints, transferTopicName, stageArtifactsRoot)
+	if err != nil {
+		err = fmt.Errorf("transfer error, cause: `%w`", err)
+		fmt.Printf("Sending error message to Teams\n")
+		teamsErrorMessage := teams.CreateGeneralFailureMessage(err)
+		if err := teams.SendMessage(teamsErrorMessage, common_flags.TeamsWebhookNotificationUrl); err != nil {
+			err = fmt.Errorf("failed to send error message to Teams, cause: `%w`", err)
+			fmt.Printf("%s\n", err)
+		}
+		return err
+	}
+	return nil
```

Comment:
I argue that the program should return the original error as well as send a message to teams. This is important because when run in kubernetes as a pod in a deployment, kubernetes will crashloopbackoff the container if an error occured but will restart the container indefinitely without a delay if the program exited with error code 0.


```suggestion
	return err
```

Reply:
My bad, my intent was to return the error message, but for some reason I did not. Fixed in https://github.com/nlnwa/hermetic/pull/23/commits/c02afc6baaf37d389b563024aa4370b5ee5565cb